harness/unittests/test_google_adapter.py

```
@@ -0,0 +1,163 @@
+        self.assertEqual(config.max_output_tokens, 42)
+        self.assertIsInstance(config.thinking_config, _FakeThinkingConfig)
+        self.assertEqual(config.thinking_config.thinking_budget, 77)
+        self.assertIsInstance(config.system_instruction, _FakeContent)
+        self.assertEqual(config.system_instruction.parts[0].text, self.google_module.SYS_PROMPT)
+
+    def test_adapter_does_not_retry_on_parameter_errors(self) -> None:
+        """Verify we surface parameter errors immediately without retries."""
+        adapter = self.google_module.GoogleAdapter(model="gemini-2.5-pro-test")
+        _FakeModels.error_to_raise = TypeError("unexpected keyword argument 'config'")
+
+        with self.assertRaises(TypeError):
```

Ruff (0.14.3)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)